homeassistant/components/elmax/cover.py

```diff
@@ -0,0 +1,137 @@
+
+    # Immediately run a discovery, so we don't need to wait for the next update
+    _discover_new_devices()
+
+
+class ElmaxCover(ElmaxEntity, CoverEntity):
+    """Elmax Cover entity implementation."""
+
+    _attr_supported_features = (
+        CoverEntityFeature.OPEN | CoverEntityFeature.CLOSE | CoverEntityFeature.STOP
+    )
+
+    def __check_cover_status(self, status_to_check: CoverStatus) -> bool | None:
+        """Check if the current cover entity is in a specific state."""
+        state = self.coordinator.get_cover_state(self._device.endpoint_id).status
+        if state is None:
+            return None
+        return state == status_to_check
+
+    @property
+    def is_closed(self) -> bool | None:
+        """Tells if the cover is closed or not."""
+        return self.coordinator.get_cover_state(self._device.endpoint_id).position == 0
+
+    @property
+    def current_cover_position(self) -> int | None:
+        """Return current position of cover.
+
+        None is unknown, 0 is closed, 100 is fully open.
+        """
+        return self.coordinator.get_cover_state(self._device.endpoint_id).position
+
+    @property
+    def is_opening(self) -> bool | None:
+        """Tells if the cover is opening or not."""
+        return self.__check_cover_status(CoverStatus.UP)
+
+    @property
+    def is_closing(self) -> bool | None:
+        """Return if the cover is closing or not."""
+        return self.__check_cover_status(CoverStatus.DOWN)
+
+    async def async_stop_cover(self, **kwargs: Any) -> None:
+        """Stop the cover."""
+        # To stop the cover, Elmax requires us to re-issue the same command once again.
+        # To detect the current motion status, we request an immediate refresh to the coordinator
+        await self.coordinator.async_request_refresh()
+        motion_status = self.coordinator.get_cover_state(
+            self._device.endpoint_id
+        ).status
+        if motion_status is None:
+            _LOGGER.warning("Could not retrieve motion_status for elmax cover")
+            return
+        command = _COMMAND_BY_MOTION_STATUS[motion_status]
+        if command:
+            await self.coordinator.http_client.execute_command(
+                endpoint_id=self._device.endpoint_id, command=command
+            )
+        else:
+            _LOGGER.warning("Ignoring stop request as the cover is IDLE")
+
+    async def async_open_cover(self, **kwargs):
+        """Open the cover."""
+        await self.coordinator.http_client.execute_command(
+            endpoint_id=self._device.endpoint_id, command=CoverCommand.UP
+        )
+
+    async def async_close_cover(self, **kwargs):
+        """Close the cover."""
+        await self.coordinator.http_client.execute_command(
+            endpoint_id=self._device.endpoint_id, command=CoverCommand.DOWN
+        )
+
+
+# Maps the stop command to use for every cover motion status
+_COMMAND_BY_MOTION_STATUS = {
```

Comment:
I think you should move this to just after `_LOGGER`

Reply:
Done in 64b1d06fd17b21a96bcf56d2d16cedabfeb31723.